packages/external-db-mssql/lib/connection_provider.js

```diff
@@ -39,15 +39,15 @@ const init = async (cfg, _poolOptions) => {
     const poolOptions = _poolOptions || {}
 
     const _pool = new ConnectionPool(Object.assign({}, config, extraOptions(), poolOptions))
-    const pool = await _pool.connect()
 
+    const {pool, cleanup} = await _pool.connect().then((res) => {return {pool: res, cleanup: async () => await pool.close() }}).catch((e) => { return {pool:_pool, cleanup: () => { }} })
```

Comment:
1. line cannot be that long, please add some new lines next time.
2. i don't understand what happens in the case it's not connecting
3. why do we return cleanup and never use it later on ?

Reply:
2. in the case it's not connecting, we get back the unconnected pool and empty cleanup, instead of throwing an error, later on we'll use the unconnected pool in the operations (try to connect again and then show the appropriate error), and when we try to send a request we'll get a "connection closed" error.
3. we use the cleanup - we returning it in the last line, I'm returning the empty cleanup because if we're trying to close the unconnected pool it's throwing an error.

Reply:
operations cannot be a part of the connection flow, it just checks health of the pool

Reply:
After the connection is failed from any reason it's throwing an error, if I don't want the app will crash I need to return something from the conection_provider (connection, cleanup), so I thought it would make the most sense to return the unconnected pool, because when we're querying this pool is throwing appropriate error and not 'query is not a function' or something similar.

Reply:
we can also create a synthetic pool that implements a query function (that will throw an error in case it's called)

Reply:
@noam-almog I created, tell me if that's what you meant

Reply:
looks good, fix my last warning and merge